apps/roam/src/components/settings/AsyncQuerySettings.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from "react";
+import { Switch, FormGroup, Classes } from "@blueprintjs/core";
+import { OnloadArgs } from "roamjs-components/types";
+
+export function AsyncQuerySettings({
+  extensionApi,
+}: {
+  extensionApi: OnloadArgs["extensionAPI"];
+}) {
```

Comment:
Is a whole separate component required for a single toggle?  Will this be used in multiple places?

Maybe, instead, let's try to match the patterns used on the existing settings page.
The query pages has a `Checkbox` example.  But if you'd like to use a `Switch` instead you could check `DiscourseNodeCanvasSettings`.  In both places you can see how the `Label` is used to try to keep it as consistent as possible.